scalding-core/src/main/scala/com/twitter/scalding/DescribedExecutionApp.scala

```diff
@@ -0,0 +1,57 @@
+package com.twitter.scalding
+
+sealed abstract class ArgOptions
+
+object RequiredArg extends ArgOptions
+object OptionalArg extends ArgOptions
+object ListArg extends ArgOptions
```

Comment:
If going this route I think the API would be cleaner if these were case classes extending ArgOptions, taking key and description.